src/ngx_http_lua_kong_log_handler.c

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2019-2023 Kong Inc.
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *    http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ngx_http_lua_kong_common.h"
+
+
+void
+ngx_http_lua_kong_error_handler(ngx_http_request_t *r, u_char **buf, size_t *len)
```

Comment:
We might not need a patch for this function. Here is what it could be done:

1. In `POST_READ` phase, store `r->log_hander` into `ctx->original_log_hander`. The type should be a [ngx_http_log_handler_pt](https://lxr.nginx.org/source/xref/nginx/src/http/ngx_http.h?r=9021%3A8d0753760546&fi=ngx_http_log_handler_pt#ngx_http_log_handler_pt).
2. In here, do:
```C
p = ctx->original_log_hander;

/* remaining code that appends request ID */
```

Does this seems to work? This way we might be able to do it with 0 patches to Nginx core which is always better.

Reply:
It means that we will replace the original `r->log_handler` to our own handler.

Reply:
I think @dndx means to store the original log handler (in ctx) and then replace it with a function that executes the original + additional code which appends the request ID. If that work that would in fact require no additional patching. I will test that.

Reply:
here's my proposal, following @dndx's suggestion above (it works!) --> https://github.com/Kong/lua-kong-nginx-module/pull/65/commits/d185ee8abb8032e6cc2b75c4a3b97abddcdc71a1